### PR TITLE
Change checkpoints to do first-fit allocation

### PR DESCRIPTION
**Pull request by keithbostic**
@michaelcahilll, for your review.

### CODE REVIEW

**Comment by agorrod** [comment]
@keithbostic I understand why this change might be helpful. Do you have a use case that drove it?

File: src/block/block_compact.c [hunk 1]
```diff
@@ -19,15 +19,10 @@ __wt_block_compact_start(WT_SESSION_IMPL *session, WT_BLOCK *block)
...
-	 * might avoid bugs in the future.
-	 */
-	__wt_spin_lock(session, &block->live_lock);
-	block->allocfirst_save = block->allocfirst;
-	block->allocfirst = 1;
-	__wt_spin_unlock(session, &block->live_lock);
+	/* Switch to first-fit allocation. */
+	__wt_block_configure_first_fit(block, 1);
+
+	block->compact_pct_tenths = 0;
```
Comment (agorrod) [inline src/block/block_compact.c]: Is this change just about being paranoid that compact_end was never called?

Comment (keithbostic) [inline src/block/block_compact.c]: Yes, pretty much -- it should be cleared, but can't hurt to be sure.

**Comment by keithbostic** [comment]
No explicit use case -- @michaelcahill and I were talking the last time we were together, and it came up as a way to make files shrink more quickly (currently, checkpoints go at the end of the file in most cases, and those checkpoint blocks are hard to get rid of).

**Comment by michaelcahill** [comment]
OK, let's see how this goes in testing: lgtm.

**Pull request merged**